service/create/asg_stack.go

```diff
@@ -246,16 +257,55 @@ func (s *Service) updateASGStack(input asgStackInput) error {
 		Client: input.clients.CloudFormation,
 
 		// Settings.
-		ASGMaxSize:  input.asgSize,
-		ASGMinSize:  input.asgSize,
-		ImageID:     imageID,
-		Name:        key.AutoScalingGroupName(input.cluster, input.asgType),
-		TemplateURL: templateURL,
+		ASGMaxSize:            input.asgSize,
+		ASGMinSize:            input.asgSize,
+		ImageID:               imageID,
+		MaxBatchSize:          getMaxBatchSize(input.asgSize),
+		MinInstancesInService: getMinInstancesInService(input.asgSize),
+		Name: key.AutoScalingGroupName(input.cluster, input.asgType),
+		RollingUpdatePauseTime: rollingUpdatePauseTime,
+		TemplateURL:            templateURL,
+	}
+
+	stackUpdated, err := stack.Update()
+	if err != nil {
+		return stackUpdated, microerror.Mask(err)
+	}
+
+	return stackUpdated, nil
+}
+
+// getMaxBatchSize calculates the max batch size for the rolling update policy.
+func getMaxBatchSize(asgSize int) int {
+	var batchSize int
+
+	switch {
+	case asgSize <= 2:
+		batchSize = 1
+	case asgSize <= 4:
+		batchSize = 2
+	default:
+		// Calculate batch size and round up to nearest int.
+		result := float64(asgSize) * asgMaxBatchSizeRatio
+		batchSize = int(result + 0.5)
 	}
 
-	if err := stack.Update(); err != nil {
-		return microerror.Mask(err)
+	return batchSize
+}
+
+// getMinInstancesInService calculates the min number for instances to keep in
+// service for the rolling update policy.
+func getMinInstancesInService(asgSize int) int {
+	var minInstances int
+
+	switch {
+	case asgSize <= 2:
+		minInstances = 1
+	default:
+		// Calculate min instances and round up to nearest int.
+		result := float64(asgSize) * asgMinInstancesRatio
+		minInstances = int(result + 0.5)
```

Comment:
> To confirm the operator isn't changing the TPO / CRO.

I understood this. My point is the rolling update should change the TPO. IMO the node IDs should disappear/change when the underlying node disappears/changes. This is the funny thing. It does not happen, but should happen, but when it happens it should not be done by the operator itself. 

+1 on using the provider specific primtives. AFAIK we already agreed on this multiple times. I am only wondering how this matches with our desired operator design. I do not see how this can match and play along together which is why I am talking here.